securedrop_client/queue.py

```diff
@@ -64,21 +66,26 @@ def _process(self, exit_loop: bool) -> None:
                 # priorities are processed in the order that they were submitted
                 # _by the user_ to the queue.
                 self.queue.put_nowait((priority, job))
+                self.pause()
             except ApiInaccessibleError:
                 # This is a guard against #397, we should pause the queue execution when this
                 # happens in the future and flag the situation to the user (see ticket #379).
                 logger.error('Client is not authenticated, skipping job...')
             finally:
                 session.close()
 
-            if exit_loop:
-                return
+    def pause(self):
+        self.queue.enqueue(PauseQueueJob())
+
+    def resume(self):
+        self.process()
 
 
 class ApiJobQueue(QObject):
     # These are the priorities for processing jobs.
     # Lower numbers corresponds to a higher priority.
     JOB_PRIORITIES = {
+        PauseQueueJob: 11,
```

Comment:
I think it would be cleaner if we use a priority of 10 for `InvalidTokenJob` and keep this at 11 rather than relying on `order_number`. I think `order_number` makes more sense for things like `SendReplyJob`s because they are actually the same type of job where order matters.